sdk/eventhub/Azure.Messaging.EventHubs/src/Amqp/AmqpConnectionScope.cs

```diff
@@ -0,0 +1,394 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Diagnostics;
+using System.Globalization;
+using System.Net;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core;
+using Azure.Messaging.EventHubs.Core;
+using Microsoft.Azure.Amqp;
+using Microsoft.Azure.Amqp.Framing;
+using Microsoft.Azure.Amqp.Sasl;
+using Microsoft.Azure.Amqp.Transport;
+
+namespace Azure.Messaging.EventHubs.Amqp
+{
+    /// <summary>
+    ///   Defines a context for AMQP operations which can be shared amongst the different
+    ///   client types within a given scope.
+    /// </summary>
+    ///
+    internal class AmqpConnectionScope : IDisposable
```

Comment:
This type is intended to be the primary manager of connections and associated objects.   There are some constructs within that aren't fully utilized yet, since we're only working against the management path.  We'll continue to build on this for producer and consumer needs and I'll do a final cleanup pass for any dead items when those are complete.